src/sentry/static/sentry/app/views/groupDetails/actions.jsx

```diff
@@ -35,37 +37,45 @@ class DeleteActions extends React.Component {
     };
   }
 
-  renderDisabledDiscard = () => {
-    let {project, organization} = this.props;
-    return this.state.hooksDisabled.map(hook => hook(organization, project));
-  };
-
-  renderDiscard = () => {
-    return (
-      <DropdownLink caret={true} className="group-delete btn btn-default btn-sm">
-        <li>
-          <LinkWithConfirmation
-            title={t('Discard')}
-            message={t(
-              'Discarding this event will result in the deletion ' +
-                'of most data associated with this issue and future ' +
-                'events being discarded before reaching your stream. ' +
-                'Are you sure you wish to continue?'
-            )}
-            onConfirm={this.props.onDiscard}
-          >
-            <GuideAnchor type="text" target="delete_discard" />
-            <span>{t('Delete and discard future events')}</span>
-          </LinkWithConfirmation>
-        </li>
-      </DropdownLink>
-    );
-  };
+  renderDiscardModal = ({Body, closeModal, renderDisabled, hasFeature, features}) => (
+    <React.Fragment>
+      <Body>
+        {!hasFeature && (
+          <FeatureDisabled alert featureName="Discard and Delete" feature={features[0]} />
+        )}
+        {t(
+          'Discarding this event will result in the deletion ' +
+            'of most data associated with this issue and future ' +
+            'events being discarded before reaching your stream. ' +
+            'Are you sure you wish to continue?'
+        )}
+      </Body>
+      <div className="modal-footer">
+        <Button onClick={closeModal}>{t('Cancel')}</Button>
+        <Button
+          style={{marginLeft: space(1)}}
+          priority="primary"
+          onClick={this.props.onDiscard}
+          disabled={!hasFeature}
+        >
+          {t('Discard Future Events')}
+        </Button>
+      </div>
+    </React.Fragment>
+  );
+
+  openDiscardModal = () =>
+    openModal(modalProps => (
+      <Feature
+        organization={this.props.organization}
+        features={['discard-groups']}
+        renderDisabled={p => this.renderDiscardModal({...p, ...modalProps})}
```

Comment:
Do we need this? It's mostly duplicate code. Doesn't `children` renderProp get passed `renderDisabled` too?

Reply:
Okay, with #10340 this is slightly more reasonable, it now does exactly what children does, but I added a comment as to why.

Reply:
Now, after #10340, I'm wondering why conflate the hook with `renderDisabled`? Maybe the hook is the children render prop since it has everything it needs to render 1) `renderDisabled`, 2) `children`, 3) any additional components as well